### DISCUSSION TITLE
.cubemap file generation

### DISCUSSION THREAD

**Discussion by StelyanBaltov** [jnsmalm/pixi3d#100] [General]
Hello again sir, I have yet another question about cubemaps this time. Can you tell me how to create those .cubemap files you have used in your examples (specular.cubemap and diffuse.cubemap). Also can I use a simple .png or a .jpeg for cubemap lighting environment? Also is it possible to Tween a cubemap environment lighting? 

Cheers

**Comment by jnsmalm** [comment]
Hi!

1. Go to https://polyhaven.com/hdris and download the environment you want (needs to be in HDR format), I usually choose 2K - it's probably enough.
2. Go to https://ibl.pixi3d.org and drop the file in that window
3. Click the download button, it will generate a zip containing everything you need.

Tweening between environments is not supported.

**Reply by StelyanBaltov** [reply]
Thanks for the reply. Can I create ImageBasedLighting from custom standard cubemap png? Not only from hdr to .cubemap files.

**Reply by jnsmalm** [reply]
You can create a cube map object using fromFaces https://api.pixi3d.org/classes/Cubemap.html#fromFaces. There you can specify the textures/png's you want to use, one for each side/face.

```
let diffuseCubemap = Cubemap.fromFaces({
  negx: PIXI.Texture.from(...),
  posx: PIXI.Texture.from(...),
...
})

let specularCubemap = Cubemap.fromFaces({
  negx: PIXI.Texture.from(...),
  posx: PIXI.Texture.from(...),
...
})

let ibl = new ImageBasedLighting(diffuseCubemap, specularCubemap)
```
